sdk/nodejs/__tests__/eventQueue.spec.ts

```diff
@@ -381,4 +386,46 @@ describe('EventQueue Unit Tests', () => {
             fourthPublishPayloads.reduce((val: number, payload) => val + payload.events.length, 0)
         ).toBe(100)
     })
+
+    it('should not queue user event if user event and aggregate event queue exceeds max event queue size', () => {
+        const logger = dvcDefaultLogger()
+        logger.warn = jest.fn()
+
+        const eventQueue = new EventQueue(logger, 'envKey')
+        const user = new DVCPopulatedUser({ user_id: 'user1' })
+        const aggEvent = { type: EventTypes.variableEvaluated, target: 'key' }
+
+        for (let i = 0; i < 500; i++) {
+            eventQueue.queueAggregateEvent(new DVCPopulatedUser({ user_id: `user${i}` }), aggEvent, config)
+        }
+
+        for (let i = 0; i < 501; i++) {
+            eventQueue.queueEvent(user, { type: 'test_event' }, config)
+        }
+
+        expect(logger.warn).toBeCalledTimes(1)
+        expect(logger.warn).toBeCalledWith(expect.stringContaining('Max event queue size reached, dropping event'))
```

Comment:
any way to assert that the events weren't actually added here?

you can access private properties of the class if you override the type warning

Reply:
done